timelock-api/src/test/java/com/palantir/atlasdb/timelock/api/ConjureLockRequestTest.java

```diff
@@ -0,0 +1,152 @@
+    private static final OldConjureLockRequest OLD_CONJURE_LOCK_REQUEST = OldConjureLockRequest.builder()
+            .requestId(REQUEST_ID)
+            .lockDescriptors(ImmutableSet.of(LOCK_1, LOCK_2, LOCK_3, LOCK_4))
+            .acquireTimeoutMs(ACQUIRE_TIMEOUT_MS)
+            .clientDescription(CLIENT_DESCRIPTION)
+            .build();
+    private static final ConjureLockRequest DEFAULT_CONJURE_LOCK_REQUEST = ConjureLockRequest.builder()
+            .requestId(REQUEST_ID)
+            .lockDescriptors(ImmutableList.of(LOCK_1, LOCK_2, LOCK_3, LOCK_4))
+            .acquireTimeoutMs(ACQUIRE_TIMEOUT_MS)
+            .clientDescription(CLIENT_DESCRIPTION)
+            .build();
+
+    private static final byte[] BYTES_OLD = PtBytes.toBytes("old");
+    private static final byte[] BYTES_NEW = PtBytes.toBytes("new");
+    private static final byte[] BYTES_DELETED = PtBytes.toBytes("deleted");
+    private static final byte[] BYTES_CREATED = PtBytes.toBytes("created");
+    // The checksum on this does not have to be correct since the integrity check is performed by the server and not
+    // by Conjure itself
+    private static final ConjureLockRequestMetadata CONJURE_LOCK_REQUEST_METADATA = ConjureLockRequestMetadata.of(
+            ImmutableMap.of(
+                    0,
+                    ConjureChangeMetadata.unchanged(ConjureUnchangedChangeMetadata.of()),
+                    1,
+                    ConjureChangeMetadata.updated(
+                            ConjureUpdatedChangeMetadata.of(Bytes.from(BYTES_OLD), Bytes.from(BYTES_NEW))),
+                    2,
+                    ConjureChangeMetadata.deleted(ConjureDeletedChangeMetadata.of(Bytes.from(BYTES_DELETED))),
+                    3,
+                    ConjureChangeMetadata.created(ConjureCreatedChangeMetadata.of(Bytes.from(BYTES_CREATED)))),
+            ConjureLockDescriptorListChecksum.of(0, Bytes.from(PtBytes.toBytes("test-checksum-value"))));
+
+    // AtlasDB (Client) serializes and TimeLock (Server) deserializes ConjureLockRequest objects.
+    // These are the respective mappers used internally by Conjure.
+    private static final ObjectMapper SERIALIZATION_MAPPER =
+            ObjectMappers.newClientObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    private static final ObjectMapper DESERIALIZATION_MAPPER =
+            ObjectMappers.newServerObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    // This mapper is used to ensure that two JSONs are equal excluding indentation
+    private static final ObjectMapper VERIFYING_MAPPER = new ObjectMapper();
+
+    @Test
+    public void oldConjureLockRequestSerializesDeserializes() {
+        assertSerializedEquals(OLD_CONJURE_LOCK_REQUEST, OLD_CONJURE_LOCK_REQUEST_FILE);
+        assertDeserializedEquals(OLD_CONJURE_LOCK_REQUEST_FILE, OLD_CONJURE_LOCK_REQUEST, OldConjureLockRequest.class);
+    }
+
+    // Requires excludeEmptyOptionals = true
```

Comment:
It might be worth removing the `as` and to return the `assertThat`, to allow for these types of cases.

Reply:
I think an improved comment is fine. Trying to get the assert here is especially ugly for `assertSerializedEquals` because we re-read the serialized form using a different mapper (so we can ignore any indentation differences).